openvdb/points/AttributeArray.h

```diff
@@ -1063,6 +1298,53 @@ TypedAttributeArray<ValueType_, Codec_>::deallocate()
 }
 
 
+#if OPENVDB_ABI_VERSION_NUMBER >= 6
+template<typename ValueType_, typename Codec_>
+bool
+TypedAttributeArray<ValueType_, Codec_>::valueTypeIsFloatingPoint() const
+{
+    // std::is_floating_point evaluates to false for half so use std::is_integral
+    return !std::is_integral<typename VecTraits<ValueType>::ElementType>::value;
+}
+
+
+template<typename ValueType_, typename Codec_>
+bool
+TypedAttributeArray<ValueType_, Codec_>::valueTypeIsScalar() const
+{
+    return  !this->valueTypeIsVector() &&
+            !this->valueTypeIsQuaternion() &&
+            !this->valueTypeIsMatrix();
+}
```

Comment:
If we introduce a any new array type this will need to be tracked. Maybe just use is_arithmetic<> (and check is_same<char>)?

Reply:
The only reason I didn't do that is to be able to support half, which isn't included in std::is_floating_point. Question is whether an unknown type should be scalar or non-scalar - perhaps non-scalar makes more sense, I'm not sure?

Reply:
I mean ideally you'd use is_scalar here to reflect the function name but it depends on your use case. If really all you want is to return true for ints and floats then I'd use is_arithmetic with char and half with an is_same

Reply:
I think really the information I'd like to know is whether the value type is a POD type like int, half, double, etc or a struct-based type like vec3s, mat4d, etc. I agree with you that it's better if the implementation here aligns more with the C++ traits, so I've changed this to valueTypeIsClass() which overlays std::is_class.

Note that half has to be explicitly excluded, though it has been raised that the type traits there should be better specialized - https://github.com/openexr/openexr/issues/253.